cypress-io/cypress
# fix: Refetch latest cloud data when opening cloud connect modals

- Closes #23538 

### User facing changelog
* Fixes an issue where outdated organization and project information could be shown in dialogs to connect a project to the Cypress Dashboard.

### Additional details
* Adding a refetch when opening any of the Cloud Connect modals (Create/Select organization, Create/Select project) to reflect any org/project changes that have occurred outside the Cypress app since the user last logged in/opened the app
* Fixed an issue in the Select Project dialog that would continue to use old data in select menus after an updated `gql` prop was passed in

### Steps to test
This procedure involves creating/deleting cloud data. We point to Prod for cloud data by default now, if you would be more comfortable testing against Staging you can do so by launching like so `CYPRESS_INTERNAL_ENV=staging yarn dev`

1. Open a project that is not connected to the cloud (no `projectId` config entry) in the app, CT or E2E
2. Ensure you are logged in.
3. Open the dashboard in your browser (`dashboard.cypress.io` or `dashboard-staging.cypress.io` depending on how you launched). Make note of your organization memberships and what projects are in each org.
4. In the app, when mousing over the "Latest runs" and "Average duration" header tooltips you should get a "Connect project" action. Click to open the connect project modal, verify list of orgs and projects matches what is in the dashboard. Close the modal without taking any action.
5. In the dashboard, create a new project and name it something distinctive
6. In Cypress, re-open the Connect Project modal. Verify newly-created project is reflected in the options. Close the modal without taking any action
7. In the dashboard, create a new organization and name it something distinctive
8. In Cypress, re-open the Connect Project modal. Verify newly-created organization is reflected in the options.

**[Regression Test]**
9. Verify select menus within the modal update appropriately when selecting different organizations (projects list should change, previous project selection should be wiped out).
10. Select an org and a project and save selection using "Connect project" button. Verify correct project ID is saved into project config.

### How has the user experience changed?

https://user-images.githubusercontent.com/11482842/187937181-4e4daa5b-37a5-4198-a9a7-8c90ec6071d1.mov


### PR Tasks
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
